src/mono/mono/mini/simd-intrinsics-netcore.c

```diff
@@ -804,18 +886,23 @@ emit_x86_intrinsics (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature
 			}
 			return emit_simd_ins_for_sig (cfg, klass, OP_SSE_SHUFFLE, args [2]->inst_c0 /*mask*/, arg0_type, fsig, args);
 		}
-		case SN_MoveHighToLow:
-			return emit_simd_ins_for_sig (cfg, klass, OP_SSE_MOVEHL, -1, arg0_type, fsig, args);
-		case SN_MoveLowToHigh:
-			return emit_simd_ins_for_sig (cfg, klass, OP_SSE_MOVELH, -1, arg0_type, fsig, args);
-		case SN_UnpackLow:
-			return emit_simd_ins_for_sig (cfg, klass, OP_SSE_UNPACKLO, -1, arg0_type, fsig, args);
-		case SN_UnpackHigh:
-			return emit_simd_ins_for_sig (cfg, klass, OP_SSE_UNPACKHI, -1, arg0_type, fsig, args);
-		case SN_Store:
-			return emit_simd_ins_for_sig (cfg, klass, OP_SSE_STORE, 1 /*alignment*/, arg0_type, fsig, args);
-		case SN_StoreAligned:
-			return emit_simd_ins_for_sig (cfg, klass, OP_SSE_STORE, 16 /*alignment*/, arg0_type, fsig, args);
+		case SN_ConvertScalarToVector128Single:
+			if (fsig->params [1]->type == MONO_TYPE_I4)
+				return emit_simd_ins_for_sig (cfg, klass, OP_XOP_X_X_I4, SIMD_OP_SSE_CVTSI2SS, 0, fsig, args);
+			else if (fsig->params [1]->type == MONO_TYPE_I8)
+				return emit_simd_ins_for_sig (cfg, klass, OP_XOP_X_X_I8, SIMD_OP_SSE_CVTSI2SS64, 0, fsig, args);
+			else
+				g_assert_not_reached ();
+			break;
+		case SN_ReciprocalScalar:
+		case SN_ReciprocalSqrtScalar:
+		case SN_SqrtScalar:
+			if (fsig->param_count == 1)
+				return emit_simd_ins_for_sig (cfg, klass, OP_XOP_X_X, info->instc0, arg0_type, fsig, args);
+			else
+				return NULL;
```

Comment:
Possibly:
```c
// process_bb:
		case OP_XOP_X_X_X:
		case OP_XOP_X_X_I4:
		case OP_XOP_X_X_I8: {
			LLVMValueRef args [] = { lhs, rhs };
			IntrinsicId id = (IntrinsicId)0;
			switch (ins->inst_c0) {
			case SIMD_OP_SSE_CVTSI2SS: id = INTRINS_SSE_CVTSI2SS; break;
			case SIMD_OP_SSE_CVTSI2SS64: id = INTRINS_SSE_CVTSI2SS64; break;
			case SIMD_OP_SSE_CVTSI2SD: id = INTRINS_SSE_CVTSI2SD; break;
			case SIMD_OP_SSE_CVTSI2SD64: id = INTRINS_SSE_CVTSI2SD64; break;
			case SIMD_OP_SSE_MAXPS: id = INTRINS_SSE_MAXPS; break;
			case SIMD_OP_SSE_MAXSS: id = INTRINS_SSE_MAXSS; break;
			case SIMD_OP_SSE_MINPS: id = INTRINS_SSE_MINPS; break;
			case SIMD_OP_SSE_MINSS: id = INTRINS_SSE_MINSS; break;
			default: goto scalar_two_arg;
			}
			values [ins->dreg] = call_intrins (ctx, id, args, "");
			break;

			scalar_two_arg:
			switch (ins->inst_c0) {
			case SIMD_OP_SSE_RCPSS: id = INTRINS_SSE_RCP_SS; break;
			case SIMD_OP_SSE_RSQRTSS: id = INTRINS_SSE_RSQRT_SS; break;
			case SIMD_OP_SSE_SQRTSS: id = INTRINS_SSE_SQRT_SS; break;
			default: g_assert_not_reached (); break;
			}
			LLVMValueRef ix[4] = { const_int32 (4), const_int32 (1), const_int32 (2), const_int32 (3) };
			LLVMValueRef mixed = LLVMBuildShuffleVector (builder, lhs, rhs, LLVMConstVector (ix, 4), "");
			values [ins->dreg] = call_intrins (ctx, id, &mixed, "");
			break;
		}
// ...
// emit_x86_intrinsics
		case SN_ReciprocalScalar:
		case SN_ReciprocalSqrtScalar:
		case SN_SqrtScalar:
			// ...
			else if (fsig->param_count == 2) {
			    return emit_simd_ins_for_sig (cfg, klass, OP_XOP_X_X_X, info->instc0, arg0_type, fsig, args);
			} else
			// ...
```

Using [sqrtss as an example](https://godbolt.org/z/drKaaH):
```
declare <4 x float> @llvm.x86.sse.sqrt.ss(<4 x float>)

@sink = global <4 x float> zeroinitializer, align 16

define void @test(<4 x float> %upper, <4 x float> %value) {
  %mixed = shufflevector <4 x float> %upper, <4 x float> %value, <4 x i32> <i32 4, i32 1, i32 2, i32 3>
  %result = tail call <4 x float> @llvm.x86.sse.sqrt.ss(<4 x float> %mixed)
  store <4 x float> %result, <4 x float>* @sink, align 16
  ret void
}
```
will yield, with LLVM 6:
```
test:                                   # @test
        movss   xmm0, xmm1              # xmm0 = xmm1[0],xmm0[1,2,3]
        sqrtss  xmm0, xmm0
        movaps  xmmword ptr [rip + sink], xmm0
        ret
```
The intermediate `movss` is optimized away in LLVM 7+.